docs/user-guide/connectors/custom-connectors.rst

```diff
@@ -31,6 +31,28 @@ To send a message, you would run a command like:
 
 where ``myio`` is the name of your component.
 
+If you need to use extra information from your front end in your custom
+actions, you can add this information in the ``metadata`` dict of your user
+message. This information will accompany the user message through the rasa
+server into the action server when applicable, where you can find it stored in
+the ``tracker``. Message metadata will not directly affect NLU classification
```

Comment:
```suggestion
the ``tracker.latest_message``. Message metadata will not directly affect NLU classification
```
please sanity check before committing this suggestion 😄

Reply:
After tested it this was the result:
```
{'intent': {'name': 'custom_action', 'confidence': 0.9610694646835327}, 'entities': [], 'intent_ranking': [{'name': 'custom_action', 'confidence': 0.9610694646835327}, {'name': 'greet', 'confidence': 0.11322032660245895}, {'name': 'mood_unhappy', 'confidence': 0.04066481441259384}, {'name': 'deny', 'confidence': 0.023363031446933746}, {'name': 'mood_great', 'confidence': 0.0}, {'name': 'affirm', 'confidence': 0.0}, {'name': 'goodbye', 'confidence': 0.0}], 'text': 'custom action'}
```
So  `latest_message` does not have the `metadata`.

But printing the latetest event:
```
{'event': 'user', 'timestamp': 1564509965.2832844, 'text': 'custom action', 'parse_data': {'intent': {'name': 'custom_action', 'confidence': 0.9610694646835327}, 'entities': [], 'intent_ranking': [{'name': 'custom_action', 'confidence': 0.9610694646835327}, {'name': 'greet', 'confidence': 0.11322032660245895}, {'name': 'mood_unhappy', 'confidence': 0.04066481441259384}, {'name': 'deny', 'confidence': 0.023363031446933746}, {'name': 'mood_great', 'confidence': 0.0}, {'name': 'affirm', 'confidence': 0.0}, {'name': 'goodbye', 'confidence': 0.0}], 'text': 'custom action'}, 'input_channel': 'cmdline', 'message_id': 'cf5852e0aef547219c30eef1829ff43d', 'metadata': None}
```

`metadata` is there, I guess I need to update `latest_message` too :thinking:

Reply:
I think that's fine how it is actually. Latest message only has the parse data Wouldn't change that as part of this pr.

Reply:
got it. makes sense! thanks for checking